src/views/datasets/DatasetDetailView.vue

```diff
@@ -186,21 +198,33 @@ onMounted(() => {
           la source originale peuvent avoir été perdues lors de leur
           récupération. Nous travaillons actuellement à améliorer la situation.
         </div>
-        <!-- add dataset to topic (if enabled) -->
-        <div v-if="topicPageConf && userStore.loggedIn && canAddTopic">
+        <div
+          v-if="canEdit || canAddToTopic"
+          class="fr-mt-2w fr-col-auto fr-grid-row fr-grid-row--middle flex-gap"
+        >
           <DsfrButton
-            class="fr-mt-2w"
+            v-if="canEdit"
+            secondary
             size="md"
-            :label="`Ajouter à un ${topicPageConf.labels.singular}`"
-            icon="fr-icon-file-add-line"
-            @click="showAddToTopicModal = true"
-          />
-          <DatasetAddToTopicModal
-            v-if="showAddToTopicModal"
-            v-model:show="showAddToTopicModal"
-            :topic-page-key="datasetsConf.add_to_topic?.page || 'topics'"
-            :dataset="dataset"
+            label="Éditer"
+            icon="fr-icon-pencil-line"
+            @click="goToEdit"
           />
+          <!-- add dataset to topic (if enabled) -->
+          <template v-if="canAddToTopic && topicPageConf">
```

Comment:
`topicPageConf` already tested in `canAddToTopic`

```suggestion
          <template v-if="canAddToTopic">
```